src/pet.js

```diff
@@ -0,0 +1,111 @@
+const initialFitness = 10;
+const initialHunger = 0;
+
+const hungerGrowUpChange = 5;
+const hungerFeedChange = 3;
+
+const fitnessGrowUpChange = 3;
+const fitnessWalkChange = 4;
+
+const maxFitness = 10;
+const minHunger = 0;
+
+const minFitnessForWalk = 3;
+const minHungerForFeed = 5;
+
+function Pet(name) {
+  this.name = name;
+  this.age = 0;
+  this.hunger = initialHunger;
+  this.fitness = initialFitness;
+  this.children = []
+};
+
+Pet.prototype = {
+  get isAlive() {
+    return this.fitness > 0 && this.hunger < 10 && this.age < 30;
+  }
+};
+
+Pet.prototype.growUp = function() {
+  if (!this.isAlive) {
+    throw new Error('Your pet is no longer alive! :o(')
+  };
+  try {
+  } catch (e) {
+    console.error(e)
+  };
+
+  this.age += 1;
+  this.hunger += hungerGrowUpChange;
+  this.fitness -= fitnessGrowUpChange;
+
+  if (this.fitness > maxFitness) {
+    this.fitness = maxFitness;
+  };
+};
+
+Pet.prototype.walk = function() {
+  if (!this.isAlive) {
+    throw new Error('Your pet is no longer alive! :o(')
+  };
+  try {
+  } catch (e) {
+    console.error(e)
+  };
+
+  this.fitness += fitnessWalkChange;
+
+  if (this.fitness > maxFitness){
+    this.fitness = maxFitness;
+  };
```

Comment:
This could be refactored a bit. In essence you are adding a value to the existing fitness, without allowing it go over a set limit. You can do this with Math.min, which will return the lowest of the values passed to it:
```
this.fitness = Math.min(this.fitness += fitnessWalkChange, maxFitness)
```